xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml

```diff
@@ -67,8 +67,11 @@
     &lt;/div&gt;
     &lt;div class="col-xs-12 col-sm-6"&gt;
       &lt;div class="input-group"&gt;
-        &lt;input type="search" name="text" class="form-control withTip useTitleAsTip"
+        &lt;input id='search-page-bar-input' type="search" name="text" class="form-control withTip useTitleAsTip"
           title="$services.localization.render('search.page.bar.query.title')" value="$escapetool.xml($text)"/&gt;
+        &lt;label class="sr-only" for='search-page-bar-input'&gt;
+          $services.localization.render('search.page.bar.query.title')
+        &lt;/label&gt;
```

Comment:
To summarize this, I think we should either let the browser use the title as fallback or add a shorter label that is different from the title.

Reply:
> Why is this label after the input?

It's a mistake, the label--input order is reversed for the checkboxes (label after the input), which explains why I was doing it this way by default in this PR. Addressed in https://github.com/xwiki/xwiki-platform/pull/2187/commits/a49534d6132a986d32404626e9521553c820bdd3.

> Also, I'm wondering if "Enter your search query" is a good label. The whole point of "Rule 4: Avoid Browser Fallback" in https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#cardinalru is "Because the purpose of these attributes is not naming, their content typically yields low quality accessible names that are not effective.". If we just re-use the title, it seems there is little value added.

Here, I think the title content we use is already short and good for a label. 
It would be read by NVDA as "Enter your search query; edit; <value>" where <value> is nothing when first loading the page. I didn't test it myself, but checked [this page about support of input[type="search"] by screen readers](https://a11ysupport.io/tests/tech__html__input__input-search#assertion-html-input(type-search)_element-provide_shortcuts-).


We have the hand over the value that we use in the label and I think this value used in the title is of good enough quality, at least in english. Note that there's a lot of places where we already use title and label indiscriminately. We discussed it a bit with Vincent in the past, and the difference seemed too subtle to matter. So far from what I saw in xwiki-platform titles are appropriate names. If needed, a change to the title value can be done, since there is usually no strong constraints on it (we're just adding the strong constraint 'be a proper accessible name' to a loosely defined set of constraints that pretty much match our new constraint: be short, be understandable, be to the point, ...).

The value added by this change is minimal, but it ensures that all browser/AT combos react the way we want it to. The title is a common fallback but it might not be perfectly supported at all times, and labels are a way more standard alternative to naming inputs.